src/argilla/client/feedback/dataset/base.py

```diff
@@ -230,45 +175,14 @@ def __init__(
                 raise ValueError(
                     "Expected `guidelines` to be either None (default) or a non-empty string, minimum length is 1."
                 )
-        self.__guidelines = guidelines
-
-        self._records: List[FeedbackRecord] = []
-        self._new_records: List[FeedbackRecord] = []
-
-    def __len__(self) -> int:
-        """Returns the number of records in the dataset."""
-        return len(self.records)
-
-    def __getitem__(self, key: Union[slice, int]) -> Union[FeedbackRecord, List[FeedbackRecord]]:
-        """Returns the record(s) at the given index(es).
+        self._guidelines = guidelines
 
-        Args:
-            key: the index(es) of the record(s) to return. Can either be a single index or a slice.
-
-        Returns:
-            Either the record of the given index, or a list with the records at the given indexes.
-        """
-        if len(self.records) < 1:
-            raise RuntimeError(
-                "In order to get items from `rg.FeedbackDataset` you need to either add"
-                " them first with `add_records` or fetch them from Argilla or"
-                " HuggingFace with `fetch_records`."
-            )
-        if isinstance(key, int) and len(self.records) < key:
-            raise IndexError(f"This dataset contains {len(self)} records, so index {key} is out of range.")
-        return self.records[key]
+        self.records: List[FeedbackRecord] = []
```

Comment:
Should not we move this to the `FeedbackDataset` class? The `ArgillaFeedbackDataset` does not need a local list of records right?

Maybe the base class can just define an abstract property `records`, letting each implementation resolver in the proper way

```suggestion
   @property
   def records(self) -> Iterable[FeedbackRecord]:
      raise NotImplementedError()
```

Reply:
It was defined that way because we're defining the `unify_responses` and `prepare_for_training` methods in the `FeedbackDatasetBase`, but I guess we can just define a property, but note that the property itself it's not an iterable when working with local datasets, as it's just a list, but I'll figure that out 👍🏻